airflow/models.py

```diff
@@ -613,6 +614,384 @@ def __init__(self, dag):
         self.pickle = dag
 
 
+from airflow.ti_dependencies.task_instance_dependency import TIDep
+class EndDateAfterExecutionDateDep(TIDep):
```

Comment:
Agreed, this was on my TODO list and I coincidentally made this change briefly after your comment!